lib/ocpp/v201/device_model.cpp

```diff
@@ -34,61 +73,77 @@ bool DeviceModel::component_criteria_match(const Component& component,
+                return false;
+            }
+            return true;
         }
-        return true;
-    case DataEnum::dateTime:
-        return true;
-    case DataEnum::boolean:
-        return (value == "true" or value == "false");
-    case DataEnum::OptionList: {
-        // OptionList: The (Actual) Variable value must be a single value from the reported (CSV) enumeration list.
-        if (!characteristics.valuesList.has_value()) {
+        case DataEnum::dateTime: {
+            DateTime d(value);
             return true;
         }
-        const auto values_list = ocpp::get_vector_from_csv(characteristics.valuesList.value().get());
-        return std::find(values_list.begin(), values_list.end(), value) != values_list.end();
-    }
-    default: // same validation for MemberList or SequenceList
-        // MemberList: The (Actual) Variable value may be an (unordered) (sub-)set of the reported (CSV) valid values
-        // list.
-        // SequenceList: The (Actual) Variable value may be an ordered (priority, etc) (sub-)set of the reported (CSV)
-        // valid values.
-        {
+        case DataEnum::boolean:
+            return (value == "true" or value == "false");
+        case DataEnum::OptionList: {
+            // OptionList: The (Actual) Variable value must be a single value from the reported (CSV) enumeration list.
             if (!characteristics.valuesList.has_value()) {
                 return true;
             }
             const auto values_list = ocpp::get_vector_from_csv(characteristics.valuesList.value().get());
-            const auto value_csv = get_vector_from_csv(value);
-            for (const auto& v : value_csv) {
-                if (std::find(values_list.begin(), values_list.end(), v) == values_list.end()) {
-                    return false;
+            return std::find(values_list.begin(), values_list.end(), value) != values_list.end();
+        }
+        default: // same validation for MemberList or SequenceList
+            // MemberList: The (Actual) Variable value may be an (unordered) (sub-)set of the reported (CSV) valid
+            // values list. SequenceList: The (Actual) Variable value may be an ordered (priority, etc) (sub-)set of the
+            // reported (CSV) valid values.
+            {
+                if (!characteristics.valuesList.has_value()) {
+                    return true;
+                }
+                const auto values_list = ocpp::get_vector_from_csv(characteristics.valuesList.value().get());
+                const auto value_csv = get_vector_from_csv(value);
+                for (const auto& v : value_csv) {
+                    if (std::find(values_list.begin(), values_list.end(), v) == values_list.end()) {
+                        return false;
+                    }
                 }
+                return true;
             }
-            return true;
         }
+    } catch (const std::exception& e) {
+        return false;
```

Comment:
Should we print here what failed so it's at least not silent?

Reply:
Yes we could, although in this function we dont have a lot of context that we can use (no component and variable names)

Reply:
What about moving the whole try catch to the next layer then? The exception would tell something about what went wrong at least.

Reply:
Yep Ill have a look and add this